crates/driver/src/main.rs

```diff
@@ -85,13 +86,22 @@ async fn init_common_components(args: &Arguments) -> CommonComponents {
     let native_token_contract = WETH9::deployed(&web3)
         .await
         .expect("couldn't load deployed native token");
+    let tenderly_api = Some(()).and_then(|_| {
+        Some(Arc::new(
+            TenderlyHttpApi::new(
+                client.clone(),
+                args.tenderly_user.as_deref()?,
+                args.tenderly_project.as_deref()?,
+                args.tenderly_api_key.as_deref()?,
+            )
+            .expect("failed to create Tenderly API"),
```

Comment:
I think if at least 1 tenderly argument is set not supplying the other 2 should result in a hard error. Otherwise they might only supply the API key, see no message that tenderly wasn't initiated and assume everything should work.

If you want to keep 3 separate arguments it could make sense to have a `TenderlyArguments` struct (similar to `token_owner_finder::Argument`) and then have a function which initializes a `TenderlyApi` from it that has nice error handling messages which could then be shared across the current and new driver.